### ISSUE TITLE
Add tap, long tap and double tap gesture

### ISSUE THREAD

**Issue by jrmi** [pmndrs/react-use-gesture#241]
Although this 3 gestures can already be done with current existing one, proper hooks can ease the job and have consistent implementation.

**Information:**
 - React Use Gesture version: [v8.1.0-beta.6]

**Comment by dbismut** [comment]
Once 8.1.0 is released I can consider a pull request to implement this.

**Comment by Raaghu** [comment]
+1

**Comment by dbismut** [comment]
Note that `onDblClick` and `onClick` handlers exist in React + this lib has a `tap` state attribute and a `delay` config option.

All in all I guess this makes detection of any of these gestures trivial IMO.

**Comment by tim-soft** [comment]
I created a useDoubleClick hook for a project for this very reason, since onDblClick will still fire your onClick handler. Really simple  code
https://github.com/tim-soft/use-double-click

**Comment by kevhcha** [comment]
+1 Would love a double tap as well!

**Comment by dbismut** [comment]
I don't think I'll handle this myself. If someone wants to give it a shot please open a PR.

**Comment by mindyourlifeguide** [comment]
+1 Would love a double tap as well!

**Comment by laclance** [comment]
yeah I have that problem, added my own long press handler but now useDrag 's down variable is getting stuck on true, so dragging happens after long press even without clicking again

**Comment by dbismut** [comment]
@laclance would you mind elaborating a bit more and possibly submit a sandbox?

**Comment by laclance** [comment]
So the problem if you press or hold after swiping then a swipe will always get triggered at the same time. This was not the case until I upgraded from react-use-gesture.

https://codesandbox.io/s/laughing-einstein-23kd2

**Comment by dbismut** [comment]
@laclance thanks for the sandbox. Can you describe what you're trying to achieve feature-wise? The code seems a bit complex at first glance.

**Comment by laclance** [comment]
sorry I tried to remove as much code as I could, so basically my our app has many objects that have difference functionality, when pressed, long press or swiped (e.g select, edit, delete, respectively, with animations) and also switch screen by swiping when on mobile.

**Comment by laclance** [comment]
it's been a while since I created the hooks but I believe most of the complexity came about when trying to get the events working on multiple objects, for example when the container and items within the container all need press, hold and swipe events.

**Comment by dbismut** [comment]
@laclance Have a look at this https://codesandbox.io/s/great-tu-kzc8m?file=/src/App.tsx

**Comment by laclance** [comment]
thank you, I ended up just implemented everything myself, wasn't to difficult but that might be helpful if I need more animations.

**Comment by mrsaleh** [comment]
It's still open ? after 4 years , we really need tap and double tap

**Comment by say8425** [comment]
```ts
import { useDoubleTap } from 'use-double-tap';

const bind = useDoubleTap((event) => {
  console.log('Double tapped');
});

useGesture(
  {
    onClick: (event) => {
      console.log('click');

      bind.onClick(event.event as unknown as MouseEvent);
    },
  }
)
```

I don't know how to implement longpress, but double tab can be implemented by binding the [use-double-tap](https://minwork.gitbook.io/double-tap-hook/basics) to `onClick`.
